### PR TITLE
Create World Inequality Database dataset

### PR DESCRIPTION
**Pull request by paarriagadap**
Create World Inequality Database dataset from a file extracted from the `wid` command in Stata.

### CODE REVIEW

**Comment by paarriagadap** [comment]
I wanted to include a notebook documenting data issues, but it generates an error in the checks. Is it OK to merge with the Jupyter notebook then?

**Review by pabloarosado — CHANGES_REQUESTED** [review]
Was the notebook added by mistake? I think we decided to move away from notebooks in ETL, as a general rule.
In any case, I don't think it's very useful to print out all data in a notebook, it becomes a really big file, and really hard to read. So:
1. My general advise would be to not have printed data in any repos. But if you think this notebook is useful for the public to visit and read through it, maybe you could move it to the notebooks repos, and add the link to that repos here in ETL somewhere in the documentation of the data step. I'd also suggest to not print out all data, I don't think that's useful for you or for an external visitor. Instead you could just print a few illustrative examples.
2. If instead this notebook is meant to be internal, with the purpose of running sanity checks (for the next time you run an update, for example), then it should not be a notebook. It could simply be a .py file (using jupytext, for example) which runs a bunch of assertions. It can also print things and plot charts and do everything you need to perform those checks. But since it's a .py file, all that printed data would not live in ETL.

Do you agree? Thanks.

**Comment by paarriagadap** [comment]
Oh, I did it like that because we discussed the possibility with you and Mojmir of including notebook files in a meeting some weeks ago for analysis purposes, and I understood that was possible. If not, I can leave it in a jupytext file.

File: etl/steps/data/garden/wid/2023-01-27/shared.py
Comment (pabloarosado) [inline etl/steps/data/garden/wid/2023-01-27/shared.py:182]: I understand that you are defining `new_line` just for convenience, in case you decide to change the number of line breaks, right? The explanation sounds as if there's some technical reason why you need to define the line break this way (cause you could simply add `\n\n` inside the f-string and it should be fine, as far as I know).
In any case, given that this is a global constant a the module level, you should name it with capital letters, e.g. `NEW_LINE`.

File: etl/steps/data/garden/wid/2023-01-27/issues.py
Comment (pabloarosado) [inline etl/steps/data/garden/wid/2023-01-27/issues.py:80]: General advice (maybe not for all cases): If you want to check things, it may be better to use assertions, like `assert variable >0, "Unexpected value for variable"`. Otherwise, you will need to read through lots of prints, and that will simply take you more time. So, print only if something is unexpected, otherwise don't print anything.

File: etl/steps/data/grapher/wid/2023-01-27/world_inequality_database.py
Comment (pabloarosado) [inline etl/steps/data/grapher/wid/2023-01-27/world_inequality_database.py:29]: Here you could do `ds_grapher = create_dataset(dest_dir, tables=[tb_garden], default_metadata=ds_garden.metadata)` and delete the next line of code.

File: etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py
Comment (pabloarosado) [inline etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py:28]: Here I think `create_dataset` will not work, since `formats` cannot be specified, so keep it as it is now. I'll create an issue.

**Review by pabloarosado — APPROVED** [review]
It looks good. I made some suggestions, but feel free to ignore them if you think they are not necessary. Another piece of general advice would be to use more descriptive variable names. Things like `dis` could easily be `disposable` and `nat` could be `national`. If it doesn't make the lines unreasonably long, descriptive names make your future reading easier (as well as working with the dataset for any other person loading it).

File: etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py
Comment (pabloarosado) [reply etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py:28]: Actually, I've just created [another PR](https://github.com/owid/etl/pull/826) that should fix the issue. So, once it's merged, you can git pull changes from master to your current branch, and then reduce the code to simply:
```
ds_explorer = create_dataset(dest_dir, tables=[tb_garden], default_metadata=ds_garden.metadata, formats=["csv"])
ds_explorer.save()
```
(and don't forget to import `create_dataset`). But if this is too much hassle, you can ignore it.

File: etl/steps/data/meadow/wid/2023-01-27/world_inequality_database.py
Comment (paarriagadap) [reply etl/steps/data/meadow/wid/2023-01-27/world_inequality_database.py:193]: Thanks, maybe it could be added like that as a default on `walkthrough` (if it's not already)

Comment (paarriagadap) [reply etl/steps/data/meadow/wid/2023-01-27/world_inequality_database.py:193]: Ah, [it is](https://owid.slack.com/archives/C0193RW5E2J/p1675688494455699)! Thank you very much then!

File: etl/steps/data/garden/wid/2023-01-27/world_inequality_database.py
Comment (paarriagadap) [reply etl/steps/data/garden/wid/2023-01-27/world_inequality_database.py:18]: Yeah, there are many. In this case, the names are given by construction in Stata, so every value containing `share` represents shares.

File: etl/steps/data/garden/wid/2023-01-27/shared.py
Comment (paarriagadap) [reply etl/steps/data/garden/wid/2023-01-27/shared.py:182]: I think it was because at the beginning I was trying to use it inside the brackets in an f-string, which breaks it.

File: etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py
Comment (paarriagadap) [reply etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py:21]: Ah, because I followed your explorer code. But I wasn't sure about that part. Should I do it like in `grapher`, then?

Comment (paarriagadap) [reply etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py:21]: Also because if I do it just like in `grapher` I have this error 
```
etl.helpers.UnknownChannel: Unknown channel name. Valid channels are 'walden', 'snapshot', 'meadow', 'garden', or 'grapher'.
```

File: etl/steps/data/garden/wid/2023-01-27/shared.py
Comment (paarriagadap) [reply etl/steps/data/garden/wid/2023-01-27/shared.py:182]: Now there's not much need to have it in a variable, so I deleted it. Thanks.

File: etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py
Comment (pabloarosado) [reply etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py:21]: You are right, PathFinder doesn't work for explorer steps (yet!) I've just created [an issue](https://github.com/owid/etl/issues/829) and I think it should be easy to fix, so wait a bit on this and it should be fixed soon. I think we should avoid using `*` and instead we should load the exact dependency specified in the dag.

File: etl/steps/data/garden/wid/2023-01-27/world_inequality_database.meta.yml
Comment (paarriagadap) [reply etl/steps/data/garden/wid/2023-01-27/world_inequality_database.meta.yml:35]: They do, but my intention is to change them to the HTML bullet points

File: etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py
Comment (pabloarosado) [reply etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py]: It was a simple fix: https://github.com/owid/etl/pull/831
Just wait for it to be merged and try again. Let me know if it does not work.

Comment (paarriagadap) [reply etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py]: Thanks! I will wait, rebase and try it

Comment (paarriagadap) [reply etl/steps/data/explorers/wid/2023-01-27/world_inequality_database.py]: Thanks, it worked.

**Pull request merged**
